pkg/scheduler/core/spreadconstraint/group_clusters.go

```diff
@@ -135,10 +135,100 @@ func groupClustersIgnoringTopology(
+	// |-----------------------------------------------------------
+	// | score            |    0    |    0    |    0    |    0    |
+	// |------------------------------------------------|---------|
+	// |AvailableReplicas |   60    |    60   |    60   |    60   |
+	// |------------------------------------------------|---------|
+
+	// According to our expectations, Group 2 is a more ideal choice than Group 1,
+	// as the number of clusters in Group 2 that meet the Replica requirements
+	// for available copies is greater. Although the average Cluster.Score in Group 1 is higher,
+	// under the Duplicate replica allocation strategy,
+	// we prioritize whether the number of available replicas in each Cluster
+	// meets the Replica requirements. Based on our algorithm, the score for Group 2
+	// is also higher than that of Group 1.
+
+	// Group1's Score = 2 * 1000 + 100 = 2100
+	// Group2's Score = 4 * 1000 + 0 = 4000
+
+	// There is another example, the rbSpec.Replicas == 50.
+
+	// There is the Group 1, it has three clusters as follows.
+	// ----------------------------------------------------------------------
+	// | clusterName      | member1 | member2 | member3 | member4 | member5 |
+	// |---------------------------------------------------------------------
+	// | score            |   100   |   100   |   100   |   100   |   100   |
+	// |------------------------------------------------|---------|---------|
+	// |AvailableReplicas |   60    |    70   |    10   |    10   |    5    |
+	// |------------------------------------------------|---------|---------|
+
+	// There is the Group 2, it has four clusters as follows.
+	// ------------------------------------------------------------
+	// | clusterName      | member1 | member2 | member3 | member4 |
+	// |-----------------------------------------------------------
+	// | score            |    0    |    0    |    0    |    0    |
+	// |------------------------------------------------|---------|
+	// |AvailableReplicas |   100   |    100  |    10   |    10   |
+	// |------------------------------------------------|---------|
+
+	// According to our expectations, Group 1 is a more ideal choice than Group 2.
+	// Although the number of clusters meeting the Replica requirements for available
+	// copies is the same in both Group 1 and Group 2, the average Cluster.Score in Group 1 is higher.
+	// Therefore, Group 1 is the better choice. Based on our algorithm,
+	// the score for Group 1 is also higher than that of Group 2.
+
+	// Group1's Score = 2 * 1000 + 100 = 2100
+	// Group2's Score = 2 * 1000 + 0 = 2000
+
+	// the priority of validClusters is higher than sumValidScore.
+	weightedValidClusters := validClusters * weightUnit
+	return weightedValidClusters + sumValidScore/validClusters
+}
+
 func (info *GroupClustersInfo) calcGroupScore(
 	clusters []ClusterDetailInfo,
 	rbSpec *workv1alpha2.ResourceBindingSpec,
 	minGroups int) int64 {
+	if checkIfDuplicate(rbSpec) {
```

Comment:
we don't need such a redundant function `checkIfDuplicate`, use `rbSpec.Placement.ReplicaSchedulingType() == policyv1alpha1.ReplicaSchedulingTypeDuplicated` instead.